Utils/utils/password_generator.py

```diff
@@ -0,0 +1,29 @@
+import secrets
+import string
+from sekoia_automation.action import Action
+
+
+class PasswordGenerator(Action):
+    def run(self, arguments: dict):
+        password_length = self.json_argument("password_length", arguments)
+        number_of_digits = self.json_argument("number_of_digits", arguments)
+        number_of_special_characters = self.json_argument("number_of_special_characters", arguments)
+
+        letters = string.ascii_letters
+        digits = string.digits
+        special_chars = string.punctuation
+
+        alphabet = letters + digits + special_chars
+
+        while True:
+            password = ""
+            for i in range(password_length):
+                password += "".join(secrets.choice(alphabet))
+
+            if (
+                sum(char in special_chars for char in password) >= number_of_special_characters
+                and sum(char in digits for char in password) >= number_of_digits
+            ):
+                break
```

Comment:
This method is not efficient, especially when `number_of_special_characters` or `number_of_digits` are getting close to `password_length`
```suggestion
        chars = [secrets.choice(special_chars) for _ in range(number_of_special_characters)]  # n special chars
        chars += [secrets.choice(digits) for _ in range(number_of_digits)] # n digits
        chars += [secrets.choice(alphabet) for _ in range(password_length - num_special - num_digits)]  # fill remaining with random
        random.shuffle(chars)  # Shuffle it so char and digits are not always at the beginning
        password = "".join(chars)
```